src/main/java/javacake/tasks/RecurringTask.java

```diff
@@ -0,0 +1,102 @@
+package javacake.tasks;
+
+import javacake.exceptions.DukeException;
+
+import java.util.Calendar;
+import java.util.Date;
+
+public class RecurringTask extends ToDo {
+    private Date date;
+    protected Frequency frequency;
+    protected String freq;
+
+    public enum Frequency {
+        DAILY, WEEKLY, MONTHLY
+    }
+
+    /**
+     * Constructor for recurring tasks.
+     * @param description Task description
+     * @param startDateTime Starting date of task
+     * @param frequency daily, weekly or monthly
+     */
+    public RecurringTask(String description, Date startDateTime, String frequency) {
+        super(description);
+
+        this.date = startDateTime;
+
+        try {
+            setFrequency(frequency);
+        } catch (DukeException e) {
+            //e.getMessage();
+        }
```

Comment:
We went through this in class. https://nuscs2113-ay1920s1.github.io/website/schedule/week6/tutorial-cs2113.html